### PR TITLE
Gives the mime two new spells: Invisible Chair and Invisible Box. [READY]

### PR DESCRIPTION
**Pull request by harmonyn**
## About The Pull Request

- Adds two new mime spells to the game. Both work like the invisible wall, i.e. they spawn an invisible item for a limited amount of time:
- **Invisible Chair:** Summons a chair and has the mime sit in it. Perfect for crowded shuttles or staying put.
- **Invisible Box:** Everyone's favorite stock mime routine! The mime summons a box (invisible to everyone but him, although you do get the message once spawned) that can store small items. When it disappears, the items are dropped.
- To prevent massively overpowered mimes, the mime gets a spellbook at roundstart from which he can choose only one out of the three mime spells.

## Why It's Good For The Game

The mime doesn't get nearly as much cool stuff as the clown, and these are pretty common pantomime routines to riff off.

## Changelog
:cl:
add: After receiving many complaints about mimes who never pantomime, Nanotrasen has liquidated its mime personnel and hired new mimes who know more routines.
add: The mime gets the choice of two new spells: Invisible Chair and Invisible Box. They work much like the Invisible Wall spell does and disappear after a short span of time. The invisible chair can be buckled to like usual chairs; the box works like a standard inventory box item. 
tweak: Mime spells are now granted via a spellbook that starts out in the mime's inventory. Mimes can choose one of these three.
/:cl:

### CODE REVIEW

**Comment by Nickkoranda** [comment]
Reading the tator mime technique book should let you have all the mime abilities

**Comment by AarontheIdiot** [comment]
Mime buff. I like it.

**Comment by ATH1909** [comment]
> Reading the tator mime technique book should let you have all the mime abilities

Maybe it'd let you conjure chairs and boxes that won't fade away with time? The forcewall from the advance guide to mimery series is an upgraded version of the invisible wall, after all. Perhaps it should give you the "upgraded form" of whatever mime spells you know, then the "basic form" of whatever mime spells you don't know?

**Comment by zombiegums** [comment]
Very cute PR, hope it gets merged. Mimes need more love.

**Comment by Dawson1917** [comment]
Mime gang finally getting the attention it deserves

**Comment by HomocidlePancakes** [comment]
Hello I am already touhou and I approve of this PR

**Comment by bgobandit** [comment]
I’m not giving traitor mimes all the new spells, they are already kind of powerful for antags individually (especially the box) and I am considering nerfing them. plus you’re a mime not a ninja

**Comment by datorangebottle** [comment]
These...

don't sound OP in the slightest.

Just got to make a way to forcibly dismiss the box, like stunning the mime. Which would be very comical.

I see no reason why the mime can't have all of these roundstart, given how the clown has SO MUCH SHIT, and all the items involved are pretty low impact.

**Comment by ExcessiveUseOfCobblestone** [comment]
I thought the box was going to be the cardboard box and I was going to lul pretty hard.

**Comment by ATH1909** [comment]
> I’m not giving traitor mimes all the new spells, they are already kind of powerful for antags individually (especially the box) and I am considering nerfing them. plus you’re a mime not a ninja

Bump up the cost of the (super?) advanced guide to mimery, then? It'd also give you a chance to buff mime bullets, which are laughably bad (the finger gun spell gives you only three of them, and they deal only 20 damage each (LESS THAN A STETCHKIN)).

**Comment by ATH1909** [comment]
> I thought the box was going to be the cardboard box and I was going to lul pretty hard.

I'll admit, I was a bit disappointed to discover that he had remembered to set the foldable variable to false.

**Comment by Timonkeyn** [comment]
> plus you’re a mime not a ninja

have you never heard of Johnny Cage the Ninja Mime?

**Comment by bgobandit** [comment]
(btw this is not stale I am on vacation)

**Comment by bigfatbananacyclops** [comment]
let’s be honest, i think every mime will just pick wall because it’s good

**Comment by bgobandit** [comment]
>(btw this is not stale I am on vacation)

**Comment by FloranOtten** [comment]
Damnit oranges, can you read

**Comment by bgobandit** [comment]
get those WIP and Stale tags off, this is ready

File: code/modules/jobs/job_types/mime.dm [hunk 1]
```diff
@@ -45,7 +45,39 @@
...
+	dat += "<A href='byond://?src=[REF(src)];invisible_chair=1'>Invisible Chair</A><BR>"
+	dat += "<A href='byond://?src=[REF(src)];invisible_box=1'>Invisible Box</A><BR>"
+	user << browse(dat, "window=book")
+
+/obj/item/book/mimery/Topic(href, href_list)
+	..()
+	if (usr.stat || usr.restrained() || src.loc != usr)
+		return
+	if (!ishuman(usr))
+		return 1
```
Comment (optimumtact) [inline code/modules/jobs/job_types/mime.dm]: return 1 ?

File: code/modules/spells/spell_types/mime.dm [hunk 2]
```diff
@@ -27,6 +27,80 @@
...
+
+/obj/effect/proc_holder/spell/aoe_turf/conjure/mime_box/cast(list/targets,mob/user = usr)
+	..()
+	var/turf/T = user.loc
+	for (var/obj/item/storage/box/mime/B in T)
+		user.put_in_hands(B)
+		B.alpha = 255
+		addtimer(CALLBACK(B, /obj/item/storage/box/mime/.proc/emptyStorage, FALSE), (summon_lifespan - 1))
+
+/obj/effect/proc_holder/spell/aoe_turf/conjure/mime_box/Click()
```
Comment (optimumtact) [inline code/modules/spells/spell_types/mime.dm]: why are you overriding click?

Comment (bgobandit) [reply code/modules/spells/spell_types/mime.dm]: for the same reason the mime wall spell does

File: code/modules/jobs/job_types/mime.dm (hunk 1 above)
Comment (bgobandit) [reply code/modules/jobs/job_types/mime.dm]: I fixed this idk why the comment is still here

**Pull request merged**